implementations/rust/ockam/ockam_command/src/tcp/listener/create.rs

```diff
@@ -30,76 +26,31 @@ pub struct TCPListenerNodeOpts {
 
 impl CreateCommand {
     pub fn run(self, options: CommandGlobalOpts) {
-        let cfg = &options.config;
-        let node = extract_address_value(&self.node_opts.at).unwrap_or_else(|_| "".to_string());
-        let port = cfg.get_node_port(&node).unwrap();
-
-        let input_addr = match std::net::SocketAddr::from_str(&self.address) {
-            Ok(value) => value,
-            _ => {
-                eprintln!("Invalid Input Address");
-                std::process::exit(exitcode::IOERR);
-            }
-        };
-
-        // Check if the port is used by some other services or process
-        if !bind_to_port_check(&input_addr) {
-            eprintln!("Another process is listening on the provided port!");
-            std::process::exit(exitcode::IOERR);
-        }
-
-        connect_to(port, self, create_listener);
+        node_rpc(run_impl, (options, self))
     }
 }
 
-pub async fn create_listener(
-    ctx: Context,
-    cmd: CreateCommand,
-    mut base_route: Route,
```

Comment:
If you go to the `connect_to` function definition, you will see that the `base_route` is the route to the background node you are connecting to: `route![(TCP, format!("localhost:{}", port))]`.

We could add a method to the `Rpc` struct to return the node's address, but let's keep out of this PR. You can create a variable in your `run_impl` function like:
```rust
let node_name = extract_address_value(&command.node_opts.at)?;
let port = options.config.get_node_port(&node_name)?;
let mut base_route = route![(TCP, format!("localhost:{}", port))];
```

Reply:
> We could add a method to the Rpc struct to return the node's address, but let's keep out of this PR. You can create a variable in your run_impl function like:

If it is okay, I can create a separate PR to add the function to RPC struct because I need one more PR to complete the hacktoberfest. :sweat_smile: